### DISCUSSION TITLE
Mitmproxy certificate is not working for vscode extension marketplace in ubuntu

### DISCUSSION THREAD

**Discussion by anupsah1** [mitmproxy/mitmproxy#6794] [Q&A]
Can any one suggest me what i am missing because i have installed it properly and when i am running curl -x http://localhost:8080 https://marketplace.visualstudio.com this command in vscode terminal i am getting the result but when i am trying to install any extension from the vscode marketplace i am unable to do that

**Comment by alexbridge** [comment]
Certificates behavior documented: https://docs.mitmproxy.org/stable/concepts/certificates/ 

Maybe will be useful to someone: i created extension for VSCode to trace requests with mitmproxy: 
https://marketplace.visualstudio.com/items?itemName=abridge.http-ceptor 

I had to document some limitations, maybe this is exactly the case with your proxy issue:
> VS Code CA trust: VS Code sometimes do not trust CA certificate of mitmproxy, some images, request rejected (Dev Tools / Console errors).
- installing mitmproxy CA sometimes helps: sudo cp ~/.mitmproxy/mitmproxy-ca-cert.pem /usr/local/share/ca-certificates/mitmproxy.crt && sudo update-ca-certificates
- running VSCode with ignore certificate errors, eg code --ignore-certificate-errors .

**Comment by sekedus** [comment]
Instead of launching VS Code normally, you need to launch VS Code from a terminal session where the uppercase and lowercase proxy environment variables are explicitly defined. This ensures that the spawned child processes inherit the configuration.

**On macOS/Linux**:

```
export HTTP_PROXY="http://127.0.0.1:8080"
export HTTPS_PROXY="http://127.0.0.1:8080"
export http_proxy="http://127.0.0.1:8080"
export https_proxy="http://127.0.0.1:8080"

code
```

**On Windows (PowerShell)**:

```
$env:HTTP_PROXY="http://127.0.0.1:8080"
$env:HTTPS_PROXY="http://127.0.0.1:8080"
$env:http_proxy="http://127.0.0.1:8080"
$env:https_proxy="http://127.0.0.1:8080"

code
```

**Note**: Replace `8080` with the port mitmproxy is actively listening on.
